### PR TITLE
adding Haystack to integrations

### PR DESCRIPTION
**Pull request by TuanaCelik**
Hello from deepset. We've had a Chroma document store for Haystack 2.0 for a while and I was just told by @jeffchuber on https://github.com/chroma-core/chroma/pull/1253 to create a page here instead so here it is.

### CODE REVIEW

**Comment by jeffchuber** [comment]
@TuanaCelik hello! thanks for opening this PR. looking pretty good. 

> It provides tooling for all steps of the NLP application building life-cycle. This includes tooling to prepare and embed documents into vector databases, as well as building fully customizable LLM pipelines such as retrieval-augmented generative (RAG) pipelines.

this feels a little generic, can we add a list of specific tasks or things that Haystack can help a developer with? I think that will really help devs figure out if haystack will help them.

File: docs/integrations/haystack.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+slug: /integrations/haystack
+title: 💙 Haystack
+---
+
+[Haystack](https://github.com/deepset-ai/haystack) is an open-source LLM framework in Python. It provides tooling for all steps of the NLP application building life-cycle. This includes tooling to prepare and embed documents into vector databases, as well as building fully customizable LLM pipelines such as retrieval-augmented generative (RAG) pipelines.
```
Comment (TuanaCelik) [inline docs/integrations/haystack.md:6]:
```suggestion
[Haystack](https://github.com/deepset-ai/haystack) is an open-source LLM framework in Python. It provides [embedders](https://docs.haystack.deepset.ai/v2.0/docs/embedders), [generators](https://docs.haystack.deepset.ai/v2.0/docs/generators) and [rankers](https://docs.haystack.deepset.ai/v2.0/docs/rankers) via a number of LLM providers, tooling for [preprocessing](https://docs.haystack.deepset.ai/v2.0/docs/preprocessors) and data preparation, connectors to a number of vector databases including Chroma and more. Haystack allows you to build custom LLM applications using both components readily available in Haystack and [custom components](https://docs.haystack.deepset.ai/v2.0/docs/custom-components). Some of the most common applications you can build with Haystack are retrieval-augmented generation pipelines (RAG), question-answering and semantic search.
```

Comment (TuanaCelik) [reply docs/integrations/haystack.md:6]: @jeffchuber What do you think of this suggestion? It goes into a bit more detail and the specifics of what we mean.

**Comment by TuanaCelik** [comment]
@jeffchuber what do you think of the update? Do you think we are ready to merge this?

**Pull request merged**